saleor/graphql/order/mutations/draft_order_create.py

```diff
@@ -331,9 +328,56 @@ def _commit_changes(
                 order=instance, user=info.context.user, app=app
             )
 
-        instance.save(
-            update_fields=["billing_address", "shipping_address", "updated_at"]
-        )
+    @classmethod
+    def _update_shipping_price(
+        cls,
+        info: ResolveInfo,
+        instance: models.Order,
+        cleaned_input,
+        shipping_channel_listing,
+    ) -> bool:
+        if shipping_address := cleaned_input.get("shipping_address"):
+            instance.shipping_address = shipping_address
+        if (
+            instance.shipping_method
+            and instance.shipping_address
+            and instance.is_shipping_required()
+        ):
+            shipping_method_data = convert_to_shipping_method_data(
+                instance.shipping_method,
+                shipping_channel_listing,
+            )
+            manager = get_plugin_manager_promise(info.context).get()
+            error = get_shipping_method_availability_error(
+                instance, shipping_method_data, manager
+            )
+            if error:
+                raise ValidationError({"shipping_method": error})
+            instance.base_shipping_price = shipping_channel_listing.price
+            return True
+        else:
+            instance.base_shipping_price = zero_money(instance.currency)
+            return False
+
+    @classmethod
+    def _get_shipping_channel_listing(cls, instance, cleaned_input):
+        if shipping_method := cleaned_input["shipping_method"]:
+            shipping_channel_listing = (
+                shipping_models.ShippingMethodChannelListing.objects.filter(
+                    shipping_method=shipping_method, channel=instance.channel
```

Comment:
If you don't have prefetched `channel`, AFAIk this will raise an additional DB query for fetching a channel. I would suggest to use `channel__id=instance.channel_id`

Reply:
Thanks for catching!